src/uniprotkb/components/entry/similar-proteins/SimilarProteins.tsx

```diff
@@ -24,70 +26,70 @@ import {
   UniRefLiteAPIModel,
 } from '../../../../uniref/adapters/uniRefConverter';
 
+type IsoformsAndCluster = {
+  isoforms: string[];
+  cluster: UniRefLiteAPIModel;
+};
+
+type Mapping = Record<UniRefEntryType, IsoformsAndCluster[]>;
+
 type Props = {
   isoforms: { isoforms: string[] };
   primaryAccession: string;
 };
 
 const SimilarProteins = ({ isoforms, primaryAccession }: Props) => {
   const allAccessions = useMemo(
-    () => [primaryAccession, ...isoforms.isoforms],
+    () =>
+      [primaryAccession, ...isoforms.isoforms].filter(
+        // remove first isoform, it'll be the primary accession
+        (accession) => !accession.endsWith('-1')
+      ),
     [primaryAccession, isoforms]
   );
 
-  const searchUrl = getClustersForProteins(allAccessions);
-
-  // Get the clusters in which the canonical and isoforms are found
-  const { loading, data, error } = useDataApi<{
-    results: UniRefLiteAPIModel[];
-  }>(searchUrl);
-
-  const clusterData = useMemo(() => {
-    if (data) {
-      const { results } = data;
-      // Remove all items with only 1 member as it will be canonical/isoform
-
-      const filtered = results
-        .map((item) => ({
-          ...item,
-          members: item.members.filter(
-            (member) =>
-              !member.startsWith('UPI') && !allAccessions.includes(member)
-          ),
-        }))
-        .filter((item) => item.members.length > 1);
-
-      if (filtered.length === 0) {
-        return null;
+  const [mappingData, setMappingData] = useSafeState<Mapping | null>(null);
+  const [mappingLoading, setMappingLoading] = useSafeState(true);
+
+  useEffect(() => {
+    setMappingLoading(true);
+    const promises = allAccessions.map((accession) =>
+      fetchData<{
+        results: UniRefLiteAPIModel[];
+      }>(`${apiUrls.search(Namespace.uniref)}?query=(uniprot_id:${accession})`)
+    );
+    Promise.all(promises).then((responses) => {
+      const mapping: Mapping = {
+        UniRef100: [],
+        UniRef90: [],
+        UniRef50: [],
+      };
+      for (const [accession, response] of zip(allAccessions, responses)) {
+        /* istanbul ignore if */
+        if (!accession || !response) {
+          break; // Shouldn't happen, used to restric types
+        }
+        for (const cluster of response.data.results) {
+          let association: IsoformsAndCluster | undefined = mapping[
+            cluster.entryType
+          ].find((association) => association.cluster.id === cluster.id);
```

Comment:
- Wasn't sure what `association` means here at first sight - what do you think about using `foundIsoformsAndCluster` or even just `found`?
- Would be a little easier to read if the `find`'s arg name wasn't `association` as this is also used in the line above.